### PR TITLE
TSD-78 TSD-76

### PR DESCRIPTION
**Pull request by esoliani**
create all component for selected school ui. Need to fix styles and layout

### CODE REVIEW

File: app/i18n/locale_en.json [hunk 1]
```diff
@@ -44,6 +44,8 @@
     "chart.title.pupil-teacher-ratio": "pupil teacher ratio",
     "chart.title.passrate-over-time": "pass rates over time",
     "chart.title.passrate-2014": "2014 pass rate",
+    "chart.title.rank-region": "REGIONAL RANK",
```
Comment (uniphil) [inline app/i18n/locale_en.json]: I think we should keep our raw strings normal cased -- we can make any text ALL CAPS with CSS.

File: app/index.html [hunk 2]
```diff
@@ -135,6 +135,7 @@
     <script src="scripts/services/metricssrv.js"></script>
     <script src="scripts/services/worldbankapi.js"></script>
 
+    <script src="scripts/directives/rank.js"></script>
```
Comment (uniphil) [inline app/index.html]: Move up to L114

File: app/scripts/directives/rank.coffee [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ # # rank
+###
+angular.module 'edudashAppDir'
+  .directive 'rank', ->
+    restrict: 'EA'
+    templateUrl: 'views/rank.html'
+    scope:
+      data: '=datasource'
+      title: '@title'
+    link: (scope, element, attrs) ->
```
Comment (uniphil) [inline app/scripts/directives/rank.coffee]: Is this temporarily commented or can it be removed?

File: app/scripts/services/worldbankapi.coffee [hunk 4]
```diff
@@ -115,7 +115,16 @@ angular.module 'edudashAppSrv'
       getTopDistricts: (filters) ->
         metric = if filters.metric is 'avg_pass_rate' then 'avg_pass_1' else 'change_13_'
         table = if filters.educationLevel is 'primary' then 'wbank.districts_primary' else 'wbank.secondary_districts'
-        schoolSql = "SELECT district_n as name, #{metric} as rate, the_geom as location FROM #{table} WHERE #{metric} IS NOT NULL ORDER BY #{metric} #{filters.order} LIMIT 5"
+        schoolSql = "SELECT district_n as name, #{metric} as rate, ST_X(ST_Centroid(the_geom)) as longitude, ST_Y(ST_Centroid(the_geom)) as latitude FROM #{table} WHERE #{metric} IS NOT NULL ORDER BY #{metric} #{filters.order} LIMIT 5"
+        $http.get(wbApiRoot, {params: { q: schoolSql, api_key: param1}})
+
+      getRank: (filters) ->
```
Comment (uniphil) [inline app/scripts/services/worldbankapi.coffee:121]: :+1: cool

File: app/scripts/services/worldbankapi.coffee [hunk 5]
```diff
@@ -115,7 +115,16 @@ angular.module 'edudashAppSrv'
       getTopDistricts: (filters) ->
         metric = if filters.metric is 'avg_pass_rate' then 'avg_pass_1' else 'change_13_'
         table = if filters.educationLevel is 'primary' then 'wbank.districts_primary' else 'wbank.secondary_districts'
-        schoolSql = "SELECT district_n as name, #{metric} as rate, the_geom as location FROM #{table} WHERE #{metric} IS NOT NULL ORDER BY #{metric} #{filters.order} LIMIT 5"
+        schoolSql = "SELECT district_n as name, #{metric} as rate, ST_X(ST_Centroid(the_geom)) as longitude, ST_Y(ST_Centroid(the_geom)) as latitude FROM #{table} WHERE #{metric} IS NOT NULL ORDER BY #{metric} #{filters.order} LIMIT 5"
```
Comment (uniphil) [inline app/scripts/services/worldbankapi.coffee:118]: Nice postgis functions to get the lag/long :+1:

File: app/views/dashboard.html [hunk 6]
```diff
@@ -250,65 +255,34 @@ <h4 translate="legend.ptratio">Pupil-Teacher Ratio</h4>
...
-                  <img ng-if="selectedSchool.change_13_14 > 0" ng-src="images/arrowgreen.png" class="arrow">
-                  <img ng-if="selectedSchool.change_13_14 < 0" ng-src="images/arrowred.png" class="arrow">
-                  <img ng-if="selectedSchool.change_13_14 == 0" ng-src="images/arrowyellow.png" class="arrow">
-                  <span ng-class="{
-                  'text-red': selectedSchool.change_13_14 < 0,
-                  'text-yellow': selectedSchool.change_13_14 == 0,
-                  'text-green': selectedSchool.change_13_14 > 0}">
-                    {{ selectedSchool.change_13_14 }}%
-                  </span>
+                    <img ng-if="selectedSchool.change_13_14 > 0" ng-src="images/arrowgreen.png" class="arrow">
```
Comment (uniphil) [inline app/views/dashboard.html:262]: indentation should be 2 spaces

File: app/i18n/locale_en.json (hunk 1 above)
Comment (esoliani) [inline app/i18n/locale_en.json]: fixed

File: app/index.html (hunk 2 above)
Comment (esoliani) [inline app/index.html]: yo does not know about that :-1:

File: app/scripts/directives/rank.coffee (hunk 3 above)
Comment (esoliani) [inline app/scripts/directives/rank.coffee]: removed

**Pull request merged**